### ISSUE TITLE
[CI] [_doc] can't start with '_'  bwc tests failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#36468]
There are FullClusterRestartIT failures that failed with `mapping type name [_doc] can't start with '_'` error.

Example reproduction: `./gradlew :qa:full-cluster-restart:v6.0.0#oldClusterTestRunner   -Dtests.seed=82454A4C76256306   -Dtests.class=org.elasticsearch.upgrades.FullClusterRestartIT   -Dtests.method="testNewReplicasWork"   -Dtests.security.manager=true   -Dtests.locale=es-HN   -Dtests.timezone=America/Indiana/Vincennes   -Dtests.distribution=zip   -Dcompiler.java=11   -Druntime.java=8`

Example failure:
```
Caused by: org.elasticsearch.client.ResponseException: method [PUT], host [http://[::1]:43277], URI [/testnewreplicaswork], status line [HTTP/1.1 400 Bad Request]
07:49:25    > {"error":{"root_cause":[{"type":"invalid_type_name_exception","reason":"mapping type name [_doc] can't start with '_'"}],"type":"invalid_type_name_exception","reason":"mapping type name [_doc] can't start with '_'"},"status":400}
07:49:25    > 	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:309)
07:49:25    > 	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:294)
07:49:25    > 	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119)
07:49:25    > 	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177)
07:49:25    > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436)
07:49:25    > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326)
07:49:25    > 	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
07:49:25    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
07:49:25    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
07:49:25    > 	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
07:49:25    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
07:49:25    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
07:49:25    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
07:49:25    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
07:49:25    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
07:49:25    > 	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
07:49:25    > 	... 1 more
```

Other build failures: 
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+bwc-tests/195/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+default-distro-bwc-tests/167/console

Related PR #35930

**Comment by martijnvg** [comment]
Early 6.x ES versions don't allow `_doc` as type name, so looks like certain `FullClusterRestartIT` tests need to be skipped when testing against these early 6.x versions.

**Comment by astefan** [comment]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+default-distro-bwc-tests/170/console

**Comment by jtibshirani** [comment]
I merged #36511 to address the test failures and will keep an eye on the BWC builds.

**Comment by martijnvg** [comment]
Thanks @jtibshirani!
